docs/contributing.rst

```diff
@@ -80,6 +80,42 @@ guidelines when you're going to report a new bug.
 Writing patches
 ===============
 
+Before you can submit a change to the Scrapy code base, you need to:
+
+#. Learn the basics of Git_ and Github_
+
+#. Fork_ the Scrapy repository
+
+#. Clone_ your fork
+
+#. Configure_ upstream repository
+
+#. Install Scrapy using::
+  
+         pip install -e .
+
+   NOTE: Consider setting up a `Virtual Environment`_ before installing Scrapy. Virtual environment is used to create an isolated environment for 
+   python projects
+
+#. Then, create branches in the following way::
```

Comment:
Maybe this step (and the next) should be out of the list of steps above, since these steps belong to a different process that must be followed every time that you start working on a new feature or bug fix.

Reply:
@Gallaecio I could put it after the steps, to tell the user to install it?

Reply:
Install what? I’m confused :confused:, my comment is about branch creation, not the installation of anything:

Reply:
Okay, so I should put the branch creation after the steps. Is that correct ?